chapters/arrays.tex

```diff
@@ -295,35 +295,35 @@ \subsection{Specialized Array Constructor Functions}\doublelabel{specialized-arr
 zeros at the other places.\\ \hline
 \lstinline!diagonal(v)!
 &
 Returns a square matrix with the elements of vector v on the diagonal
 and all other elements zero.\\ \hline
-\texttt{zeros(n\textsubscript 1,n\textsubscript 2,n\textsubscript 3,...)} &
-Returns the $n_1$ x $n_2$ x $n_3$ x ... \lstinline!Integer! array with all elements equal to zero ($n_i \geq 0$).
-The function need one or more arguments, that is \lstinline!zeros()! is not legal.\\ \hline
-\texttt{ones(n\textsubscript 1,n\textsubscript 2,n\textsubscript 3,...)} &
-Return the $n_1$ x $n_2$ x $n_3$ x ... Integer array with all elements equal to one ($n_i \geq 0$).
-The function need one or more arguments, that is \lstinline!ones()! is not legal.\\ \hline
-\texttt{fill(s,n\textsubscript 1,n\textsubscript 2,n\textsubscript 3,...)} &
-Returns the $n_1$ x $n_2$ x $n_3$ x ... array with all elements equal to scalar or array expression s
-($n_i \geq 0$). The returned array has the same type as \lstinline!s!.
+\lstinline[mathescape=true]!zeros($n_{1}$, $n_{2}$, $n_{3}$, $\ldots$)! &
+Returns the $n_{1} \times n_{2} \times n_{3} \times \ldots$ \lstinline!Integer! array with all elements equal to zero ($n_{i} \geq 0$).
+The function needs one or more arguments, that is \lstinline!zeros()! is not legal.\\ \hline
+\lstinline[mathescape=true]!ones($n_{1}$, $n_{2}$, $n_{3}$, $\ldots$)! &
+Return the $n_{1} \times n_{2} \times n_{3} \times \ldots$ \lstinline!Integer! array with all elements equal to one ($n_{i} \geq 0$).
+The function needs one or more arguments, that is \lstinline!ones()! is not legal.\\ \hline
+\lstinline[mathescape=true]!fill(s, $n_{1}$, $n_{2}$, $n_{3}$, $\ldots$)! &
+Returns the $n_{1} \times n_{2} \times n_{3} \times \ldots$ array with all elements equal to scalar or array expression $s$
+($n_{i} \geq 0$). The returned array has the same type as \lstinline!s!.
 Recursive definition:
-\lstinline!fill!(s,$n_1$,$n_2$,$n_3$,...) =
-\lstinline!fill!(\lstinline!fill!(s,$n_2$,$n_3$, ...),
-$n_1$,); \lstinline!fill!(s,n)=\{s,s,\ldots{}, s\}
+\lstinline[mathescape=true]!fill(s, $n_{1}$, $n_{2}$, $n_{3}$, $\ldots$)! =
+\lstinline[mathescape=true]!fill(fill(s, $n_{2}$, $n_{3}$, $\ldots$), $n_{1}$)!;
+\lstinline[mathescape=true]!fill(s, $n$)! = \lstinline[mathescape=true]!{s, s, $\ldots$, s}!
 
 The function needs two or more arguments; that is \lstinline!fill(s)! is not legal.\\ \hline
-\lstinline!linspace(x1,x2,n)!
+\lstinline[mathescape=true]!linspace(x1, x2, $n$)!
 &
-Returns a \lstinline!Real! vector with \lstinline!n! equally spaced elements, such that
-$v = \text{\lstinline!linspace(x1, x2, n)!}$ results in
-\lstinline!v[i] = x1 + (x2-x1)*(i-1)/(n-1)! for $1 \leq \text{\lstinline!i!} \leq \text{\lstinline!n!}$.
-It is required that $\text{\lstinline!n!} \geq 2$. The arguments \lstinline!x1! and \lstinline!x2! shall
+Returns a \lstinline!Real! vector with $n$ equally spaced elements, such that
```

Comment:
I'm not sure if mathescape is needed for n in linspace, and to me it would it would make sense that x1, x2, and n are all treated the same. For fill it is different as we need subscripts for n_2, but that does not apply here.

Reply:
The problem is that once we start using mathescape in some places, things look inconsistent if we don't use it in other similar places as well.  Also, by using mathescape for things that represent numbers and that may be part of math in the text, the math gets much cleaner if the number-variables are treated as math everywhere.

Reply:
I agree that we need to be consistent, and to me the issue is that x1 and n are treated differently. I understand that x1 is a Real scalar and n an Integer scalar, but to me it doesn't make sense to differentiate them - and I would prefer the rule that only arguments that need math-formatting like n_1 should use that.

Reply:
I was about to just remove the math styling of _n_ when I had a look at the entire table.  To me, the thing that stands out as ugly is that the `x1` and `x2` aren't using subscripts.  Don't you agree that it would start looking odd with `n` in code style when actually most other arguments in this table would be in math style?

To me, using math style just because we want subscripts is a rather poor argument; it makes it feel like just a quick and dirty workaround for not being able to have subscripts in code snippets.  By instead applying math style to things that might be useful to also refer to as math embedded in the normal text, we give ourselves a better excuse for sometimes switching to math inside the code snippets, and I must say I actually prefer the kind of separation this introduces between the function being defined and the symbolic arguments to which it is applied — there is a lightness to the math font that helps placing the emphasis on the function name.  (I'm not going as far as asking that we should use math for formal arguments in function definitions, but I do think it would look even better.)

Reply:
To me changing font that much has some drawback, but we could view it as "meta-variable" syntax and use math-variables for all - including x1 and x2. A slight drawback is that we also explain it using Modelica-syntax and that looks odd with math-variables.
So I can live with it at least.

Reply:
Fixed.

Fortunately, many things can be explained with math syntax instead of Modelica syntax.  When it comes to inequalities, there is actually a subtle benefit when doing this (apart from looks), and that is that mathematical inequalities don't need to be wrapped in `noEvent`.